src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -71,14 +75,46 @@ public async Task<AuthenticationResult> ExecuteAsync(
             return await handler.RunAsync(cancellationToken).ConfigureAwait(false);
         }
 
+        private void ValidateAndConfigureMtlsPopForAcquireTokenForClient(
+            AcquireTokenForClientParameters clientParameters,
+            AcquireTokenCommonParameters commonParameters,
+            RequestContext requestContext)
+        {
+            if (clientParameters.UseMtlsPop)
+            {
+                // Validate that the certificate is not null for MTLS PoP
+                if (_confidentialClientApplication.Certificate == null)
```

Comment:
You need to check the authority type - all these checks are for AAD. Consider throwing exception if authority type != AAD